torch/csrc/utils/python_arg_parser.h

```diff
@@ -667,6 +671,8 @@ inline std::vector<int64_t> PythonArgs::intlistWithDefault(
       } else {
         try {
           res[idx] = THPUtils_unpackIndex(obj);
+        } catch (std::runtime_error& e) {
```

Comment:
Differntiation between a runtime error and exception to differentiate between the overflow vs type exception sounds bad?
Should the one below be just about type error?
And we need a new generic handler for "failed to convert to a valid int"?